include/util/indexed_data.hpp

```diff
@@ -346,7 +346,7 @@ template <typename GroupBlockPolicy, storage::Ownership Ownership> struct Indexe
         const GroupBlockPolicy block;
         block.ReadRefrencedBlock(blocks[block_idx], internal_idx, first, last);
 
-        return adapt(&*first, &*last);
```

Comment:
For empty container, `*first` and `*last` can both be dereferencing the end iterator.